tools/azsdk-cli/docs/specs/non-azure-models-support-proposal.md

```diff
@@ -0,0 +1,216 @@
+5. **Model Availability**: Some specialized models may only be available through specific providers
+
+### Default Recommendation
+
+**Use Microsoft Foundry (default configuration)** unless you have a specific need for an external provider AND are certain that no confidential data will be processed.
+
+## Potential Issues and Considerations
+
+### 1. Authentication Scope Hardcoding
+- **Issue**: The authentication scope `https://cognitiveservices.azure.com/.default` is hardcoded
+- **Risk**: This scope is specific to Azure Cognitive Services and may not be appropriate for all scenarios
+- **Consideration**: Future enhancements might need to make this configurable
+
+### 2. Security Considerations
+- **API Key Exposure**: Environment variables can be exposed in logs or error messages
+- **Mixed Authentication**: When both API key and Azure credentials are available, API key takes precedence
+- **Recommendation**: Document security best practices for credential management
+
+### 3. Confidential Data Leakage Risk
+- **Issue**: When using non-Microsoft model providers, there is a risk of accidentally sending Microsoft confidential data (e.g., unannounced TypeSpec APIs still in development, internal service specifications, or other proprietary information) to external parties
+- **Risk**: Sensitive information could be exposed to third-party model providers who may use the data for training or other purposes, potentially violating confidentiality agreements or revealing unreleased product features
+- **Mitigations**:
+  - **Explicit opt-in mechanism**: Require users to explicitly opt-in to using non-Azure endpoints via a CLI flag (e.g., `--allow-external-provider`) to ensure intentional use
+  - **Warning prompts**: Display a warning message when a non-Azure endpoint is detected, reminding users about data confidentiality considerations
+  - **Environment variable safeguards**: Consider requiring a specific environment variable (e.g., `AZSDK_ALLOW_EXTERNAL_AI_PROVIDERS=true`) to enable non-Azure providers
+  - **Documentation**: Clearly document the risks of using external providers with confidential data and provide guidance on appropriate use cases
+
+## Scope and Focus: Azure SDK CLI/MCP and Non-Azure SDKs
+
+### Primary Purpose
+
+The Azure SDK CLI/MCP is designed primarily for **Azure SDK development workflows**. Its core functionality, including code generation, API review, testing utilities, and development automation, is tailored specifically for the Azure SDK ecosystem.
+
+### Support for Non-Azure SDKs
+
+While the non-Azure model support enables technical flexibility, the question arises whether supporting non-Azure SDKs (such as the OpenAI .NET SDK) is desirable or meaningful within this tool.
+
+#### Considerations
+
+1. **Shared Technology Stack**: The OpenAI .NET SDK is TypeSpec-based, similar to Azure SDKs, which means many Azure SDK CLI/MCP workflows (TypeSpec generation, code generation patterns) could be applicable
```

Comment:
How the library is generated is neither a value nor a risk.   It is simply an implementation detail.   The burden of being fully compatible is on the Azure Foundry endpoints and is not impacted by whether or how the client library is generated.